hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1156,8 +1184,31 @@ protected Option<String> inlineScheduleCompaction(Option<Map<String, String>> ex
    */
   protected HoodieWriteMetadata<O> logCompact(String logCompactionInstantTime, boolean shouldComplete) {
     HoodieTable table = createTable(config, context.getHadoopConf().get());
+    try {
+      // Transaction serves to ensure only one logcompact job for this instant will start heartbeat, and any other concurrent
+      // logcompact job will abort if they attempt to execute logcompact before heartbeat expires
```

Comment:
there is lot of duplicated code b/w regular compaction and log compaction. 
may be in a follow up PR, we should try to dedup the code